fili-core/src/test/groovy/com/yahoo/bard/webservice/web/filters/RoleBasedAuthFilterSpec.groovy

```diff
@@ -15,49 +17,61 @@ import javax.ws.rs.core.SecurityContext
 @Timeout(30)
 class RoleBasedAuthFilterSpec extends Specification {
 
-    JerseyTestBinder jtb
+    private final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance()
 
-    Class<?>[] getResourceClasses() {
-        [RoleBasedAuthFilterSpec.class, DataServlet.class]
-    }
+    ObjectMapper objectMapper
+    ContainerRequestContext containerRequestContext
+    SecurityContext securityContext
+    RoleBasedAuthFilter roleBasedAuthFilter
 
     def setup() {
-        // Create the test web container to test the resources
-        jtb = new JerseyTestBinder(getResourceClasses())
+        objectMapper = Mock(ObjectMapper)
+        containerRequestContext = Mock(ContainerRequestContext)
+        securityContext = Mock(SecurityContext)
+        containerRequestContext.getSecurityContext() >> securityContext
+
+        roleBasedAuthFilter = new RoleBasedAuthFilter(objectMapper)
+        SYSTEM_CONFIG.setProperty(SYSTEM_CONFIG.getPackageVariableName("user_roles"), "foo,bar");
     }
 
     def cleanup() {
-        // Release the test web container
-        jtb.tearDown()
+        SYSTEM_CONFIG.clearProperty(SYSTEM_CONFIG.getPackageVariableName("user_roles"))
```

Comment:
We can't assume that this was unset when the test started. We need to capture the value of the property (as a `String`) in the `setup`, and then use that captured value to `reset(capturedValue)` in the cleanup.